templates/zerver/app/left_sidebar.html

```diff
@@ -4,20 +4,22 @@
             {# Special-case this link so we don't actually go to page top. #}
             <li class="top_left_all_messages top_left_row active-filter" title="{{ _('All messages') }} (Esc)">
                 <a href="#">
-                    <span class="filter-icon">
-                        <i class="fa fa-home" aria-hidden="true"></i>
-                    </span>
-                    {#- squash whitespace -#}
-                    <span>{{ _('All messages') }}</span>
-                    <span class="count">
-                        <span class="value"></span>
-                    </span>
+                    <div>
```

Comment:
Can you explain why you need these divs without any classes on them?   Ideally, we'd be adjusting the styling to be more similar to `stream_sidebar_row.handlebars` if the goal is consistency of behavior...

Reply:
The `<a>` tag has a number of elements inside it. The reason why the links did not work was that the `<span>` didn't extend till the end.
![Screenshot from 2019-06-15 23-38-42](https://user-images.githubusercontent.com/25329595/59554825-095f1180-8fc7-11e9-979f-b133185a36f4.png)

The `<div>` encapsulates the elements inside the `<a>` tag here so that solves the issue of "dead space".
![Screenshot from 2019-06-15 23-36-13](https://user-images.githubusercontent.com/25329595/59554852-3c090a00-8fc7-11e9-879f-d4a2411f6cbe.png)

`stream_sidebar_row.handlebars` seemed to have a different structure with a single `<a>` tag so I used the `div`s with the goal of encapsulating all elements in the top left sidebar. 
This could have been the wrong approach to fix the issue, and I would love to get a few notes on how to approach this if that is the case. 
Thanks. :)

Reply:
Instead of adding a `<div>`, how about setting `display: block` on the `<a>` with CSS?

Reply:
> Instead of adding a `<div>`, how about setting `display: block` on the `<a>` with CSS?

That worked! Thanks, @andersk